tools/appveyor.psm1

```diff
@@ -3,14 +3,81 @@ $repoRoot = Join-Path $PSScriptRoot '..'
 
 Import-Module (Join-Path $repoRoot 'build.psm1')
 
+# tests if we should run a daily build
+# returns true if the build is scheduled 
+# or is a pushed tag
+Function Test-DailyBuild
+{
+    if($env:APPVEYOR_SCHEDULED_BUILD -eq 'True')
+    {
+        return $true
+    }
+    if($env:APPVEYOR_REPO_TAG_NAME)
+    {
+        return $true
+    }
+    return $false
+}
+
+# Sets a build variable
+Function Set-BuildVariable
+{
+    param(
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [string]
+        $Name,
+
+        [ValidateNotNullOrEmpty()]
+        [Parameter(Mandatory=$true)]
+        [string]
+        $Value
+    )
+
+    if($env:AppVeyor)
+    {
+        Set-AppveyorBuildVariable @PSBoundParameters
+    }
+    else 
+    {
+        set-item env:/$name -Value $Value
+    }
+}
+
 # Emulates running all of AppVeyor but locally
 # should not be used on AppVeyor
 function Invoke-AppVeyorFull
 {
-    Invoke-AppVeyorInstall
-    Invoke-AppVeyorBuild
-    Invoke-AppVeyorTest -ErrorAction Continue
-    Invoke-AppveyorFinish
+    param(
+        [switch] $APPVEYOR_SCHEDULED_BUILD,
+        [switch] $CleanRepo
+    )
+    if($CleanRepo)
+    {
+        Start-PSCleanRepo
+    }
+
+    if($env:APPVEYOR)
+    {
+        throw "This function is to simulate appveyor not to be run from appveyor!"
```

Comment:
resolved

---
In reply to: [77050686](https://github.com/PowerShell/PowerShell/pull/2139#discussion_r77050686) [](ancestors = 77050686)